librubyfmt/src/ruby.rs

```diff
@@ -81,29 +83,82 @@ extern "C" {
     pub fn rb_id2name(id: ID) -> *const libc::c_char;
     pub fn rb_ary_entry(arr: VALUE, idx: libc::c_long) -> VALUE;
     pub fn rb_raise(cls: VALUE, msg: *const libc::c_char);
+    pub fn rb_block_call(
+        obj: VALUE,
+        method_id: ID,
+        argc: libc::c_int,
+        argv: *const VALUE,
+        block: extern "C" fn(_: VALUE, _: VALUE, _: libc::c_int, _: *const VALUE) -> VALUE,
+        outer_scope: VALUE,
+    ) -> VALUE;
 }
 
 pub fn current_exception_as_rust_string() -> String {
+    let ruby_string = unsafe { eval_str("$!.inspect") }.expect("Error evaluating `$!.inspect`");
```

Comment:
Rather than panicking, how about we return a string indicating that there was an error even trying to get the value out of Ruby. Or even better, is there a function in the Ruby C API that we can call instead of evaling?